sycl/source/detail/config.hpp

```diff
@@ -163,6 +165,31 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_DEVICE_FILTER> {
+  using BaseT = SYCLConfigBase<SYCL_DEVICE_FILTER>;
+
+public:
+  static device_filter_list *get() {
+    static bool Initialized = false;
+    static device_filter_list *FilterList = nullptr;
+
+    // Configuration parameters are processed only once, like reading a string
+    // from environment and converting it into a typed object.
+    if (Initialized) {
+      return FilterList;
+    }
+
+    Initialized = true;
```

Comment:
NIT: Please add a comment why thread-unsafe access to `Initialized` is OK here

Reply:
This variable is intended whether we already have checked env var and if so we avoid rechecking it.
If multiple threads are checking this env var, whoever wins (set this variable true) the other threads do not need to recheck the env var.

Reply:
Done.

Reply:
Not sure the current impl is thread safe.

Seems like a good use for https://en.cppreference.com/w/cpp/thread/call_once?

std::call_once and then unconditional return of FilterList.

Reply:
Thanks for the advice.
Actually, Constantin and I are saying it is not necessary to keep thread-safety here.
The reason is that there should not be a difference in env var across different threads.
During the program execution, env var can't be changed. So, whoever sets it first (or at the same time, or at different times), this function will always return the same env var configuration value.
If it was necessary, all other configurations should have done thread-safe implementation already.

Reply:
Thanks for the advice.
Actually, Constantin and I are saying it is not necessary to keep thread-safety here.
The reason is that there should not be a difference in env var across different threads.
During the program execution, env var can't be changed. So, whoever sets it first (or at the same time, or at different times), this function will always return the same env var configuration value.
If it was necessary, all other configurations should have done thread-safe implementation already.